### PR TITLE
Add workflow and keys spreadsheet

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

This PR adds generate keys workflow to generate and dump the spreadsheet and also adds spreasheet

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by iArchitSharma** [comment]
Thanks for the new workflow @Yashsharma1911 
I've added the table in Reference page - https://deploy-preview-225--bejewelled-pegasus-b0ce81.netlify.app/cloud/reference/
image

**Comment by leecalcote** [comment]
Feedback offered in today's meeting.

**Comment by leecalcote** [comment]
The expected outcome of this effort looks like a single, dedicated page that lists the default roles with their default permissions (and has a link / info tip to how a docs page on how to customize them) and that perms table should look like this - 
Screenshot 2024-03-01 at 3 57 55 PM

File: assets/csv/keys.csv
Comment (leecalcote) [inline assets/csv/keys.csv:1]: No.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: layouts/shortcodes/csvtable.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{{ $data := "" }}
+{{ $p := "csv/keys.csv" }}
```
Comment (Yashsharma1911) [inline layouts/shortcodes/csvtable.html:2]:
```suggestion
{{ $p := "data/csv/keys.csv" }}
```

File: data/csv/keys.csv [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Filter,View Filters,View all public and published filters of other team members and private of signed-in user,,,,,X,,,,Install extension,"INSERT INTO keychains (id, name, created_at, owner) VALUES ('Install extension','View Filters', NOW(), uuid_nil())",df41c45f-7c73-49c2-a055-0584fdcec1c1,"INSERT INTO keys (id, keychain_id, category, subcategory, function, description, created_at, owner) VALUES ('df41c45f-7c73-49c2-a055-0584fdcec1c1','Install extension','Catalog Management','Filter','View Filters','View all public and published filters of other team members and private of signed-in user', NOW(), uuid_nil())",X,TRUE
```
Comment (Yashsharma1911) [inline data/csv/keys.csv:5]:
```suggestion
Filters,View Filters,View all public and published filters of other team members and private of signed-in user,,,,,X,,,,Install extension,"INSERT INTO keychains (id, name, created_at, owner) VALUES ('Install extension','View Filters', NOW(), uuid_nil())",df41c45f-7c73-49c2-a055-0584fdcec1c1,"INSERT INTO keys (id, keychain_id, category, subcategory, function, description, created_at, owner) VALUES ('df41c45f-7c73-49c2-a055-0584fdcec1c1','Install extension','Catalog Management','Filter','View Filters','View all public and published filters of other team members and private of signed-in user', NOW(), uuid_nil())",X,TRUE

```

**Comment by Yashsharma1911** [comment]
Outside of assets directory it is giving issue @iArchitSharma , please see if you can fix and merge after that

**Comment by iArchitSharma** [comment]
@Yashsharma1911 we should remove csv from data directory
image

**Comment by iArchitSharma** [comment]
@Yashsharma1911 lets put the csv back to assets directory for now
i have changes location in shortcode to `assets/data/keys.csv`

**Comment by Yashsharma1911** [comment]
@iArchitSharma here are a few more things that can be improved:

1. Default permissions page description (it can be more expressive).
1. Audit and correct the incorrect checks for “User” role.
1. Update customizable info with exact doc on how to customise permission maybe step by step guide, instead of redirecting user to a page which describes “what are keychains” instead of how to customise it.

I’m moving forward with PR, please consider updating these in your next PR.

**Pull request merged**